### PR TITLE
feat: showing results pane in dashboard

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY

This PR has finally come to the end after completing these PRs:
- https://github.com/apache/superset/pull/20226
- https://github.com/apache/superset/pull/20201
- https://github.com/apache/superset/pull/20109
- https://github.com/apache/superset/pull/20170

Show `ResultsPane` in Dashboard and keep all functions as `ResultsPane` on explore page.

### Mulitple charts
https://user-images.githubusercontent.com/2016594/169540783-bec01028-ba4e-409d-a475-1e0166b522db.mov

### Results data copy to clipboard and formatted time switching

https://user-images.githubusercontent.com/2016594/171773659-568081e9-188a-4f72-a349-fc3f700ce6c1.mov

### TESTING INSTRUCTIONS
1. The functionality in ResultsPane should same between Dashboard and Explore
2. The HTTP request should not resend if chart not apply new filter on Dashboard. (for example, the results data will be cached until apply a new native filter)
3. Should correctly copy results data to clipboard both in Dashboard and Explore

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Loving this!!

Capturing a few UI/UX change requests that have come up in design feedback (CC @kasiazjc ❤️ )
* In the modal header, instead of "View Results" maybe we could get more specific, like one of these:
  * {Name} Data
  * {Name} Results
  * Chart Data: {Name}
  * ... or others?
* In the footer of the modal, can we add two buttons?
  * (left/secondary button) "Open in Explore" (or is this "Edit chart" now?)
  * (right/primary button) "Close"

**Review by kgabryje — APPROVED** [review]
LGTM!

File: superset-frontend/src/dashboard/components/SliceHeaderControls/index.tsx [hunk 1]
```diff
@@ -327,6 +330,41 @@ class SliceHeaderControls extends React.PureComponent<
...
+              modalBody={
+                <ResultsPane
+                  queryFormData={this.props.formData}
+                  queryForce={false}
+                  dataSize={20}
+                  isRequest
+                />
+              }
+              modalFooter={
+                <>
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/SliceHeaderControls/index.tsx:351]: I think that React.Fragment is not necessary

**Comment by jinghua-qa** [comment]
LGTM!

**Review by jinghua-qa — APPROVED** [review]
QA tested and LGTM

File: superset-frontend/src/dashboard/components/SliceHeaderControls/index.tsx (hunk 1 above)
Comment (zhaoyongjie) [reply superset-frontend/src/dashboard/components/SliceHeaderControls/index.tsx:351]: nice catch! Thanks!

**Pull request merged**